vsintegration/src/FSharp.Editor/LanguageService/Tokenizer.fs

```diff
@@ -925,7 +925,7 @@ module internal Tokenizer =
 
         lineData, textLinePos, contents
 
-    let inline tokenizeLine (documentKey, sourceText, position, fileName, defines, langVersion, strictIndentation, cancellationToken) =
```

Comment:
Why does this not inline anymore?

Reply:
It uses quite a lot of APIs marked ````private````:  I assumed that it was preferable to lose a bit of inlining rather than expanding the visibility of specifically private items.

It uses:
````
    let private getCachedSourceLineData
````
Which uses: 
````
     let private getFromRefreshedTokenCache
````
and these:
````
    /// An intermediate extraction of information from the token
    type private DraftTokenInfo =
        {
            Kind: LexerSymbolKind
            LeftColumn: int
            MatchedLength: int
        }

        static member Create kind (token: SavedTokenInfo) =
            {
                Kind = kind
                LeftColumn = int token.LeftColumn
                MatchedLength = int token.MatchedLength
            }

        member token.RightColumn = token.LeftColumn + token.MatchedLength - 1

    /// This is the data saved about each line.  It is held strongly while a file is open and
    /// is important for memory performance
    type private SourceLineData
        (
            lineStart: int,
            lexStateAtStartOfLine: FSharpTokenizerLexState,
            lexStateAtEndOfLine: FSharpTokenizerLexState,
            hashCode: int,
            classifiedSpans: ClassifiedSpan[],
            savedTokens: SavedTokenInfo[]
        ) =
        member val LineStart = lineStart
        member val LexStateAtStartOfLine = lexStateAtStartOfLine
        member val LexStateAtEndOfLine = lexStateAtEndOfLine
        member val HashCode = hashCode
        member val ClassifiedSpans = classifiedSpans
        member val SavedTokens = savedTokens

        member data.IsValid(textLine: TextLine) =
            data.LineStart = textLine.Start
            && let lineContents = textLine.Text.ToString(textLine.Span) in
               data.HashCode = lineContents.GetHashCode()

    type private SourceTextData(approxLines: int) =
        let data = ResizeArray<SourceLineData option>(approxLines)

        let extendTo i =
            if i >= data.Count then
                data.Capacity <- i + 1

                for j in data.Count .. i do
                    data.Add(None)

        member x.Item
            with get (i: int) =
                extendTo i
                data.[i]
            and set (i: int) v =
                extendTo i
                data.[i] <- v

        member x.ClearFrom(n) =
            let mutable i = n

            while i < data.Count && data.[i].IsSome do
                data.[i] <- None
                i <- i + 1

    /// This saves the tokenization data for a file for as long as the DocumentId object is alive.
    /// This seems risky - if one single thing leaks a DocumentId (e.g. stores it in some global table of documents
    /// that have been closed), then we leak **all** this associated data, forever.

    type private PerDocumentSavedData = ConcurrentDictionary<string list, SourceTextData>
    let private dataCache = new MemoryCache("FSharp.Editor.Tokenization")
````